osbuild/buildroot.py

```diff
@@ -14,115 +19,169 @@
 ]
 
 
+# pylint: disable=too-many-instance-attributes
 class BuildRoot(contextlib.AbstractContextManager):
+    """Build Root
+
+    This class implements a context-manager that maintains a root file-system
+    for contained environments. When entering the context, the required
+    file-system setup is performed, and it is automatically torn down when
+    exiting.
+
+    The `run()` method allows running applications in this environment. Some
+    state is persistent across runs, including data in `/var`. It is deleted
+    only when exiting the context manager.
+    """
+
+    _exitstack = None
+    _rootdir = None
+    _rundir = None
+    _vardir = None
+    _libdir = None
+    _runner = None
```

Comment:
Nit: Is there any value in having those at the class level? They are all set in `__init__`, anyway. Could also declare their type. :)

Reply:
I tried doing proper type-annotations in the `mpp` tool once, and it did not work out nicely. Thing is, the dependencies do not necessarily annotate their types, and then things just get annoying and require `Any` annotations everywhere. I decided for myself to just not try it anymore.